src/Native/Unix/System.Native/pal_networking.c

```diff
@@ -692,9 +373,6 @@ void SystemNative_FreeHostEntry(struct HostEntry* entry)
 
             case HOST_ENTRY_HANDLE_HOSTENT:
             {
-#if !HAVE_THREAD_SAFE_GETHOSTBYNAME_AND_GETHOSTBYADDR
-                free(entry->AddressListHandle);
-#endif
```

Comment:
Nit: this function seems like it could be condensed now to just:
```C#
void SystemNative_FreeHostEntry(struct HostEntry* entry)
{
    if (entry != NULL && entry->HandleType == HOST_ENTRY_HANDLE_ADDRINFO)
    {
        freeaddrinfo((struct addrinfo*)entry->AddressListHandle);
    }
}
```
but it's not a big deal.

Reply:
agreed, we should at least remove the empty case.

Reply:
In fact, I think we can remove the HandleType field and the enum as well, since HOST_ENTRY_HANDLE_ADDRINFO is the only one used now, right?

And AddressListHandle can be typed as (struct addrinfo *) instead of void*.